cli/src/pcluster/templates/slurm_builder.py

```diff
@@ -301,6 +301,22 @@ def _add_slurm_compute_fleet(self):
         if not self._condition_disable_cluster_dns():
             self.cluster_hosted_zone = self._add_private_hosted_zone()
 
+        self.terminate_compute_fleet_custom_resource = CfnCustomResource(
+            self.stack_scope,
+            "TerminateComputeFleetCustomResource",
+            service_token=self.cleanup_lambda.attr_arn,
+        )
+        self.terminate_compute_fleet_custom_resource.add_property_override("StackName", self.stack_name)
+        self.terminate_compute_fleet_custom_resource.add_property_override("Action", "TERMINATE_EC2_INSTANCES")
+        if not self._condition_disable_cluster_dns():
+            self.terminate_compute_fleet_custom_resource.add_depends_on(self.cleanup_route53_custom_resource)
+        for security_group in self.compute_security_groups.values():
+            # Control the order of resource deletion.
+            # Security groups can be deleted only after the compute nodes are terminated.
+            self.terminate_compute_fleet_custom_resource.add_depends_on(security_group)
+        # TODO: add depends_on resources from CloudWatchLogsSubstack and ComputeFleetHitSubstack?
```

Comment:
shall we remove this comment now?

Reply:
To be rigorous, we should add `dependsOn` from CloudWatch logs to custom resource (or lambda functions). We haven't seen any problem because CloudWatch logs seems to be created fast enough everytime